Documentation/bash-scripts.md

```diff
@@ -0,0 +1,52 @@
+# Bash scripts
+
+A common way to use acbuild is via a bash script. There's a common template
+that the examples for acbuild follow, and it makes cleanup a little easier. The
+template is at the end of this file, and here we explain the reasoning behind
+the parts of it.
+
+The 1st line, `#!/usr/bin/env bash`, is simply good form to make sure the
+script is executed with bash as opposed to the current shell.
+
+The 2nd line, `set -e` makes the script exit in the event of an error. If some
+command doesn't exit correctly, due to something like a typo in a command, or
+the command in `acbuild run` encountering an error, it is better to end the
+script, instead of continuing on and producing an ACI with unintended values or
+contents.
+
+The 4th through the 7th lines check if the script is being run as the root
+user. If it is not, the script prints an error and exits. This check may not be
+necessary, depending on the actions performed in the script.
+
+The 9th through 12th lines define a function that performs cleanup after the
+build is finished. More on that in a moment.
+
+The 14th line starts the build. If the build is to be started from an existing
+ACI, this line will be different.
+
+The 15th line sets up a trap to call the `acbuildend` function when the script
+exits. This will happen either if the end of the script is reached, or if the
+script is ending due to a command not exiting cleanly. The function it performs
+will take note of the exit code of the last command that was run, and then call
+`acbuild end`. If `acbuild end` exits cleanly, then it exits with the exit code
+it took note of.
+
+```bash
+#!/usr/bin/env bash
+set -e
+
+if [ "$EUID" -ne 0 ]; then
+    echo "This script uses functionality which requires root privileges"
+    exit 1
+fi
+
+acbuildend () {
+    export EXT=$?;
```

Comment:
Unless it conflicts with something else in the env, I'd just name this var `EXIT`